### ISSUE TITLE
allow auth with headers

### ISSUE THREAD

**Issue by davidszotten** [astral-sh/uv#1369]
would you be open to allowing passing auth headers, eg something along the lines of https://github.com/pypa/pip/issues/8042

**Comment by zanieb** [comment]
Thanks for the request!

For my own notes, xref to a "blocking" reason in `pip`: https://github.com/pypa/pip/pull/8078#issuecomment-657037943

**Comment by davidszotten** [comment]
yes it sounds like having this configured per _host as opposed to per _index is one solution to the security issue

**Comment by nwp90** [comment]
We need this in pip, and I'd like to point out that different auth per-host is not enough - for git repositories it's quite likely that bearer tokens would be configured *per-repository*.

**Comment by corleyma** [comment]
We use a poetry plugin to do this today.  (concerned only with indexes, in reference to https://github.com/pypa/pip/pull/8078#issuecomment-657037943).

A per host or per index option would make adopting `uv` over poetry very simple.

**Comment by lfagliano** [comment]
has anyone found a workaround for this by any chance?
